staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go

```diff
@@ -489,7 +489,7 @@ type Helper struct {
 // [KubeClient] and [DriverName] options are mandatory.
 // If the plugin will be used to publish resources, [NodeName]
 // is also mandatory.
-func Start(ctx context.Context, plugin DRAPlugin, opts ...Option) (result *Helper, finalErr error) {
+func Start(ctx context.Context, cancelMainContext context.CancelCauseFunc, plugin DRAPlugin, opts ...Option) (result *Helper, finalErr error) {
```

Comment:
having a context and then plumbing a cancel function that later is merged with the context is really confusing

Reply:
I'd be happy to provide a context only, but I don't see any way to get cancel function(that is a result of context.WithCancel* on the upper level) from it.

Reply:
Would not merging it make it better? I thought that one cancel function is better than two. Or we're talking about 2 different things here?

Reply:
That said "from a driver author perspective" - my interpretation was that the driver author then will do that in their code.

Reply:
`context.CancelCauseFunc` = `func(error)`. So what this does is adding a second callback that has the exact same signature as the existing `ErrorHandler` interface method, it just gets called for different errors and is provided through some different means (parameter instead of interface implementation).  I don't think we should be doing this.

It's also completely undocumented, which is a problem not only for DRA driver developers, but also for reviewers.